Add intro screens

abdulwd opened the pull request:
Fixes #421 

Changes: The onboarding screens with some animations have been added, the screens show top user benefits as described on the material design website [here](https://material.io/design/communication/onboarding.html#top-user-benefits-model).

![a](https://user-images.githubusercontent.com/20206262/40915182-ba1416c4-6818-11e8-9ca8-568a835cadf3.gif)

abdulwd commented:
@kelson42 I have used the logos of Wikipedia and other websites on the 2nd screen. Can we use the logos or should I remove them?

kelson42 commented:
@Popolechien This is a question for you

Popolechien commented:
We certainly can't use the Stack Exchange logo. Wikimedia logos should be ok but I would honestly question having such an animation with the Wikipedia logo as the center of the world. Don't you have something more static along the (graphic) lines of the airplane on pic number 3?

abdulwd commented:
@Popolechien Can we use the logo of TED? If you want I can disable the animation in pic 2.
Also, I can disable the fly-in animations in pic 3 if you say.

Popolechien commented:
Generally speaking I would not use any logo we do not own, except of course on top of a file to say "this is Wikipedia / TED / StackExchange content".
I'm very ok with pic 3, what I meant is that images/icons should have some sort of style consistency, so for instance you could replace pic 2 with something that has a similar style (but depicting, I don't know, a book or a globe or whatever you can find in the library you picked it from and connects with the concept of reading, or no internet).

abdulwd commented:
@Popolechien I agree that we need to have some sort of style consistency to replace pic 2 with something that has a similar style. Presently, I don't have something in my mind that connects with the concept of reading or no internet with such a style.
@mhutti1 For the time being, may I remove pic 2 so that we can merge this PR and when we have something similar to pic 3 I will add that to the intro screens.

mhutti1 commented:
ok

abdulwd commented:
@mhutti1 I have removed the second screen, have rebased the branch and all the tests are updated now. I have also refactored the package structure of the tests so that they follow that of the source. This PR can be merged now.

abdulwd commented:
@mhutti1 Please test this on API 14. I have only tested it only API 23.

mhutti1 commented on app/src/androidTestKiwix/java/org/kiwix/kiwixmobile/tests/activities/SplashActivityTest.java:
```diff
@@ -1,87 +0,0 @@
-package org.kiwix.kiwixmobile.tests.activities;
```
Is there a reason this is deleted? Can you comment it out instead? Same with a lot of the tests that have been deleted. Generally deleting tests is a bad idea.

abdulwd replied:
This has been renamed to **org.kiwix.kiwixmobile.main.MainActivityTest**.
I have not deleted any test. Instead of commenting out the tests I have fixed them.

- Just **org/kiwix/kiwixmobile/tests/activities/MainActivityTest.java** has been renamed to **org/kiwix/kiwixmobile/splash/SplashActivityTest.java** as we only needed to test the launch of SplashActivity. 
- And **org/kiwix/kiwixmobile/tests/activities/SplashActivityTest.java** has been renamed to **java/org/kiwix/kiwixmobile/main/MainActivityTest.java** as in this file we are testing **MainActivity** not **SplashActivity**.

Doing so will follow the same package structure that of the source packages. So we don't have to make the scope of many classes public unnecessarily.

I have only deleted a single file in tests which is **org/kiwix/kiwixmobile/tests/KiwixMobileActivityTest.java** and that was empty, containing absolutely no tests.

mhutti1 replied:
Ah my bad. thanks for clarifying this.

abdulwd commented:
![screenshot_2018-10-10-15-15-58-722_org kiwix kiwixmobile](https://user-images.githubusercontent.com/20206262/46729258-6836f300-cca2-11e8-8031-325a3706d4cd.png)
![screenshot_2018-10-10-15-16-52-587_org kiwix kiwixmobile](https://user-images.githubusercontent.com/20206262/46729259-68cf8980-cca2-11e8-84e2-a7036f7adb12.png)

Popolechien commented:
Pretty cool. I think it is spelled humankind (one word) and I am not 100% convinced with the use of "books", which is admittedly a lot more understandable than "zim files". 
@hayleynewton As a UX person /native English speaker: any thoughts? Would "content" work any better?

abdulwd commented:
@Popolechien I will correct the typo.
